conda-env/dev.yml

```diff
@@ -8,21 +8,21 @@ dependencies:
     # Base
     # ==================
     # NOTE: If versions are updated, also `additional_dependencies` list for mypy in `.pre-commit-config.yaml`
-    - python=3.10.10
-    - pip=23.1.2
-    - numpy >=2.0.0,<2.1.0
+    - python >=3.10,<3.14
+    - pip
+    - numpy >=2.0.0,<3
     - cartopy >=0.22.0
     - matplotlib >=3.7.1
-    - cdp=1.7.0
-    - eofs=2.0.0
-    - seaborn=0.12.2
-    # - enso_metrics=1.1.5
+    - cdp >=1.7.0
```

Comment:
@lee1043 Is `cdp` is being replaced in another PR? I noticed it is mainly used for argument parsing. This can probably done with the native Python [argparse](https://docs.python.org/3/library/argparse.html) library, which is what I did in `e3sm_diags`.

Reply:
@lee1043, python support reading parameters from a `.cfg` file but not from a python file.  It seems like the difficulty may come if you want to have config options that are lists or other python data structures.  I have a custom config parser in the MPAS-Tools package that parses config options like that but there isn't a python standard way to handle that as far as I know.

@tomvothecoder, what does e3sm_diags do?